feat: 게시물 댓글 작성하기

haansohee opened the pull request:
댓글 정보 로드해오는 시간이 넘 길어서.. activityIndicator추가했어요
ㅠㅠ
사용자 시스템 테마 light모드일때 시계, 배터리 색상 하얗게 나오는건 못했어여..

haansohee committed:
feat: 게시물 댓글 작성하기

wnsgur9137 commented on BookBuddy/BoardSearch/ViewController/BoardSearchViewController.swift:
```diff
@@ -274,6 +273,12 @@ extension BoardSearchViewController: UICollectionViewDataSource {
                 .disposed(by: cell.disposeBag)
             
+            cell.rx.commentButtonTapped
+                .subscribe(onNext: {[weak self] _ in
```
UI 업데이트는 drive() 이용하는게 좋아요

wnsgur9137 commented on BookBuddy/Home/ViewController/HomeViewController.swift:
```diff
@@ -163,6 +163,11 @@ extension HomeViewController: UICollectionViewDataSource {
         
+        cell.rx.commentButtonTapped
+            .subscribe(onNext: {[weak self] _ in
+                self?.present(UINavigationController(rootViewController: CommentViewController(postID: followingBoardInformation[indexPath.row].postID)), animated: true)
```
여기도 마찬가지로 drive 사용하는게 좋아보여요

wnsgur9137 replied:
여기도 마찬가지로 drive 사용하는게 좋아보여요

wnsgur9137 replied:
여기도 마찬가지로 drive 사용하는게 좋아보여요

haansohee committed:
fix: 코드 수정(subscribe -> drive)

haansohee committed:
fix: 주석 삭제

haansohee committed:
fix: statusBar 색상

haansohee committed:
fix: loading 화면 및 comment 관련

wnsgur9137 commented on BookBuddy/BoardSearch/View/Cell/BoardSearchViewCell.swift:
```diff
@@ -35,7 +35,7 @@ final class BoardSearchViewCell: UICollectionViewCell {
         let label = UILabel()
         label.translatesAutoresizingMaskIntoConstraints = false
-        label.text = "테스트닉네임"
+        label.text = "로딩 중"
```
라이브러리 SkeletonView 한번 참고해봐용

wnsgur9137 commented on BookBuddy/Comment/ViewController/CommentViewController.swift:
```diff
@@ -0,0 +1,224 @@
+        commentPostView.commentTextView.delegate = self
+    }
+    
+    private func addSubviews() {
```
addSubview하는 메서드나, 레이아웃 제약조건을 거는 메서드, 바인딩하는 메서드 등 메서드 위치 ViewController 모두 통일시켜서 작성하는걸까용

haansohee replied:
.,.,.,.,.,.,.,.,.네ㅠㅠ

wnsgur9137 commented on BookBuddy/Comment/ViewController/CommentViewController.swift:
```diff
@@ -0,0 +1,224 @@
+            .asDriver(onErrorJustReturn: false)
+            .drive(onNext: {[weak self] isCommentLoaded in
+                guard isCommentLoaded else { return }
+                guard let button = self?.commentPostView.commentPostButton else { return }
```
``` swift
guard isCommentLoaded,
      let button = self?.commentPostView.commentPostButton else { return }
```
이런식으로도 가능해용

haansohee replied:
옙

wnsgur9137 commented on BookBuddy/Comment/ViewController/CommentViewController.swift:
```diff
@@ -0,0 +1,224 @@
+        self.endEditingGesture?.isEnabled = true
+        if textView.text == textViewPlaceholder {
+            textView.text = nil
+            textView.textColor = .label
```
else 혹은 뒤에 로직이 더 없는 경우 나는 `guard`문 사용하는게 좋아보여용

haansohee replied:
넹

wnsgur9137 replied:
수정이 안 된 것 같아용

wnsgur9137 commented on BookBuddy/Comment/ViewController/CommentViewController.swift:
```diff
@@ -0,0 +1,224 @@
+            cell.isHiddenOption(true)
+        } else {
+            cell.isHiddenOption(false)
+            if let profileImage = commentInformation[indexPath.row].profile {
```
중첩 if문을 없애는 방법이 모가 있을지 고민해보는 것도 좋아보여용

haansohee replied:
오홍

wnsgur9137 commented on BookBuddy/Comment/ViewController/CommentViewController.swift:
```diff
@@ -0,0 +1,224 @@
+            return CGSize(width: width, height: height)
+        } else {
+            let height = calculateCellHeight(textString: commentInformation[indexPath.row].commentContent)
+            return CGSize(width: width, height: height)
```
``` swift
let height = commentInformation.isEmpty ? 300.0 : calculateCellHeight(textString: commentInformation[indexPath.row].commentContent)
```
삼항연산자를 사용하면 한 줄에 끝낼 수 있긴 한데, 오히려 가독성을 해치는 것 같으면 소희가 작성한거 그대루 사용해도 좋아용

haansohee replied:
일단 보기 좋게 할래용

wnsgur9137 commented on BookBuddy/Comment/ViewModel/CommentViewModel.swift:
```diff
@@ -0,0 +1,53 @@
+        if let uploadProfileData = UserDefaults.standard.data(forKey: UserDefaultsForkey.profile.rawValue) {
+            uploadProfile = uploadProfileData
+        } else {
+            uploadProfile = Data()
```
이런 경우
``` swift
uploadProfile = UserDefaults.standard.data(forKey: userDefaultsForkey.profile.rawValue) ?? Data()
```
를 사용할 수 없나용

haansohee replied:
할수있어용

wnsgur9137 commented on BookBuddy/Home/ViewController/HomeViewController.swift:
```diff
@@ -63,12 +59,10 @@ extension HomeViewController {
     }
     
     private func loadFollowingBoardInformations() {
+        navigationController?.present(activityIndicatorViewController, animated: false)
```
navigationController에서 present하는 이유가 뭘까요?

haansohee committed:
fix: (1) subscrive -> drive
(2) skeletonView 적용 (3) CommentViewController guard isComment 코드 변경 (4) else 뒤 로직 없는 경우 guard문으로 변경 (5) 중첩 if문 제거 (6) if let 코드를 ?? 로 언래핑으로 변경 (7) navigationController에서 present하는 코드 -> DispatchQueue.main.async로 변경

wnsgur9137 commented on BookBuddy/BookSearch/View/SearchResultCell.swift:
```diff
@@ -29,7 +30,7 @@ final class SearchResultCell: UICollectionViewCell {
         let label = UILabel()
         label.translatesAutoresizingMaskIntoConstraints = false
-        label.text = "로딩 중..."
+        label.isSkeletonable = true
```
UI 컴포넌트들은 isSkeletonable만 true값 주고 showSkeleton()은 사용하지 않고있나요?

wnsgur9137 commented on BookBuddy/Member/Edit/ViewController/MemberEditViewController.swift:
```diff
@@ -57,14 +62,16 @@ extension MemberEditViewController {
-                self?.memberEditView.profileImageView.image = UIImage(systemName: "person")
-            }
+//            DispatchQueue.main.async { [weak self] in
+//                self?.memberEditView.profileImageView.image = UIImage(systemName: "person")
```
주석처리 안지웠대용

haansohee replied:
흥

haansohee committed:
fix: (1) skeletonView showSkeleton메서드를 viewDidLoad/viewWillAppear로.. (2) 로그인파할때.. (3) 또 뭐였찌..

haansohee committed:
fix: if문 -> guard 문
